### ISSUE TITLE
[Accessibility] MatDialog is trapping tab key but not headings navigation shortcuts

### ISSUE THREAD

**Issue by opozo** [angular/material2#7787]
#### Bug, feature request, or proposal:

MatDialog is trapping tab key but not headings navigation shortcuts, this is actually working well on AngularJS Material [1] [2]

[1] https://material.angularjs.org/latest/demo/dialog
[2] https://github.com/angular/material/blob/master/src/components/dialog/dialog.js#L1134

#### What is the expected behavior?

Screen reader should trap tab key as well as headings navigation shortcuts when modal dialog is opened.

#### What is the current behavior?

Screen reader is not trapping headings navigation shortcuts when modal dialog is opened.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

"@angular/material": "2.0.0-beta.12"

**Comment by jelbourn** [comment]
@crisbeto escalating this since it came up from a team in Google. AFAIK the only way to do this would be to add `aria-hidden` to all of the non-dialog content while its open. I think we should be able to mark all of the siblings to the overlay container as `aria-hidden` (except for our own `aria-live` element and anything _already_ `aria-hidden`), keeping track so we can restore them upon close. 

We might also want to deal with stacked dialogs, but that's not as much as a priority.
